### PR TITLE
[Support TTS provider] Sarvam

### PR DESCRIPTION
**Pull request by ryzen-xp**
close #

### CODE REVIEW

**Comment by LilaRest** [comment]
Hi @ryzen-xp, thanks for the Sarvam TTS implementation! However, there are several critical issues that need to be addressed:

1. **Not implementing streaming**: This is a blocking implementation that doesn't support streaming. The Sarvam API appears to be a REST API that returns the full audio at once. According to the Life.js requirements, **streaming is mandatory** for all TTS providers. If Sarvam doesn't support streaming, this provider cannot be merged.

2. **Incorrect audio format**: You're sending base64 audio directly to the chunk (line 50):
   ```ts
   // Current (incorrect)
   voiceChunk: base64Audio,
   
   // Should be Int16Array PCM data
   const audioBuffer = Buffer.from(base64Audio, 'base64');
   const pcmBytes = new Int16Array(audioBuffer.buffer, audioBuffer.byteOffset, audioBuffer.length / 2);
   voiceChunk: pcmBytes,
   ```

3. **Missing test file**: No test file was provided. Please add a test at `packages/life/models/tts/tests/sarvam.test.ts` following the pattern in other TTS tests.

4. **Creating client on every request**: You're creating a new `SarvamAIClient` instance for each text chunk (line 24). This is inefficient and should be initialized once in the constructor.

5. **Incorrect schema export name**: The schema is exported as `sarvamSchema` but should be `sarvamTTSConfigSchema` to follow the naming convention.

6. **Missing documentation**: No provider documentation added to `apps/website/content/docs/configuration/models.mdx`.

7. **No text buffering**: Since this appears to be a blocking API, you need to buffer text until `isLast` is true (when that parameter is added to the base class). Currently, you're making an API call for every text chunk, which is inefficient and may produce choppy audio.

8. **Error handling**: Using `any` type for error (line 55). Should properly type the error.

10. **Incomplete configuration**: Missing important options like:
    - Audio format/encoding
    - Sample rate
    - Speed/pitch controls (if supported)

**Suggestions:**

If Sarvam supports WebSocket streaming:
- Implement proper WebSocket-based streaming similar to Cartesia or ElevenLabs

If Sarvam only supports REST API:
- This provider cannot be accepted as streaming is mandatory
- Consider reaching out to Sarvam to request streaming API support

**Before submitting:**
- Ensure streaming is properly implemented (mandatory requirement)
- Ensure your test passes fully and covers edge cases
- Verify the code is minimal and cannot be simplified further
- Add proper JSDoc comments to your provider class and methods
- Run `bun run types` to ensure no TypeScript errors

File: packages/life/models/tts/providers/sarvam.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+// Config
+export const sarvamTTSConfigSchema = z.object({
+  apiKey: z.string().default(process.env.SARVAM_API_KEY ?? ""),
+});
+
+// Model
+export class SarvamTTS extends TTSBase<typeof sarvamTTSConfigSchema> {
+
+  #sarvam: SarvamAIClient;
+  #socket: any;
```
[the remark this answers is not in the document]
Comment (ryzen-xp) [reply packages/life/models/tts/providers/sarvam.ts:15]: Hello GM @LilaRest  , Can you  suggest any  solution for above Typescript  error ,.

[the remark this answers is not in the document]
Comment (LilaRest) [reply packages/life/models/tts/providers/sarvam.ts:15]: Hey, try to find a Socket* or  *Socket type exported by `sarvamai` package using Intellisense suggestions. If not exported, you can build it yourself using something like `type SarvamSocket = ReturnType<SarvamAIClient.textToSpeechStreaming.connect>` (might need adjustment).
